### ISSUE TITLE
Rule proposal: `prefer-set-size`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1700]
### Description

Come up with the idea because I wrote something stupid. https://github.com/sindresorhus/globby/blob/51c8f68e63dc7a49f986bd0751e3488ced7485ec/tests/globby.js#L301

### Fail

```js
const set = new Set(foo);
const length = [...set].length;
```

```js
const uniqueLength = [...new Set(array)].length
```

### Pass

```js
const set = new Set(foo);
const {size} = set;
```

```js
const uniqueLength = new Set(array).size;
```

**Comment by sindresorhus** [comment]
Accepted
